Formatted header component

claudiulodro opened the pull request:
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request, and the reason for such changes. -->

Closes #18 .

Adds the FormattedHeader component. The code is strongly influenced by [the Calypso component of the same name.](https://github.com/Automattic/wp-calypso/blob/master/client/components/formatted-header/index.jsx) I've also made a couple tweaks to the Subscriptions demo page, for better demoing.

I'll remove the Do Not Merge label and add tests to this as soon as the tests are working again. :)

### How to test the changes in this Pull Request:

1. `npm ci` `npm run build:webpack`
2. Navigate to the Subscriptions screen.
<img width="566" alt="Screen Shot 2019-05-07 at 11 34 17 AM" src="https://user-images.githubusercontent.com/7317227/57324895-b5a30380-70bd-11e9-8268-c6117f754ca1.png">

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

<!-- Mark completed items with an [x] -->

jeffersonrabb commented on assets/src/components/formattedHeader/index.js:
```diff
@@ -0,0 +1,34 @@
+	 */
+	render() {
+		const { headerText, subHeaderText } = this.props;
+		const classes = 'newspack-formatted-header' + ( !! subHeaderText ? ' has-subheader' : '' );
```
You could use the [`classnames`](https://jedwatson.github.io/classnames/) utility for a (maybe?) more elegant approach here. In this case your approach is fewer characters and requires no dependency so it's superior, but just mentioning this for future cases where we get into complex className logic. It would look like this:

`import classnames from 'classnames';`

then...

```
const classes = classnames(
			'newspack-formatted-header',
			!! subHeaderText ? 'has-subheader' : null
		);
```

claudiulodro replied:
Yeah, I'm sure that utility will come in handy a lot, so let's just start using it. 👍

jeffersonrabb commented on assets/src/components/formattedHeader/index.js:
```diff
@@ -0,0 +1,34 @@
+			<header className={ classes }>
+				<h1 className="newspack-formatted-header__title">{ headerText }</h1>
+				{ subHeaderText && (
+					<p className="newspack-formatted-header__subtitle">{ subHeaderText }</p>
```
Why not use `h2` for  more semantic tagging?

jeffersonrabb commented:
Why the DO NOT MERGE tag?

claudiulodro commented:
I've made the requested revisions. The DO NOT MERGE tag was because I wanted to add unit tests before we merged it. I've added unit tests (and updated the branch so the tests work), so we should be good to merge.
